components/Input/src/index.svelte

```diff
@@ -1,53 +1,71 @@
 <script lang="ts">
+	import type { KInputProps } from './types';
 	import { createEventDispatcher } from 'svelte';
 	import { fade } from 'svelte/transition';
 	import { KIcon } from '@ikun-ui/icon';
-	export let iconPrefix = '';
-	export let iconSuffix = '';
-	export let value = '';
-	export let cls = '';
-	export let placeholder = '';
-	export let disabled = false;
-	export let attrs = {};
+	import { createCls, getPrefixCls } from '@ikun-ui/utils';
+
+	export let value: KInputProps['value'] = '';
+	export let placeholder: KInputProps['placeholder'] = '';
+	export let disabled: KInputProps['disabled'] = false;
+	export let iconPrefix: KInputProps['iconPrefix'] = '';
+	export let iconSuffix: KInputProps['iconSuffix'] = '';
+	export let cls: KInputProps['cls'] = '';
+	export let attrs: KInputProps['attrs'] = {};
 	/**
 	 * @internal
 	 */
-	export let isError: boolean = false;
+	export let isError: KInputProps['isError'] = false;
 	/**
 	 * @internal
 	 */
-	export let errorMsg: string = '';
+	export let errorMsg: KInputProps['errorMsg'] = '';
+
 	const dispatch = createEventDispatcher();
-	const onUpdated = (e: Event) => {
+
+	const onInput = (e: InputEvent) => {
 		if (disabled) return;
 		dispatch('input', (e.target as HTMLInputElement).value);
 	};
+
+	const onChange = (e: Event) => {
+		if (disabled) return;
+		dispatch('change', e);
+	};
+
 	const onEnter = (e: KeyboardEvent) => {
 		if (disabled) return;
 		if (e.key === 'Enter') dispatch('enter', e);
 		else dispatch('keydown', e);
 	};
-	const onChange = (e: Event) => {
-		if (disabled) return;
-		dispatch('change', e);
-	};
-	let valueInner = value;
+
+	// class names
+	const prefixCls = getPrefixCls('input');
```

Comment:
`k-input--base__dark` -> `k-input--base_dark` ?
Modifications here cause style problems in dark mode